### PR TITLE
Patch/modal

### PR DESCRIPTION
**Pull request by dammmula**
Create functionality for using modals

### CODE REVIEW

**Comment by dammmula** [comment]
[General Modal Component](https://trello.com/c/PzbBKnez/18-general-modal-component)

File: client/src/components/modals/modal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import { H3, Icon } from '@blueprintjs/core';
+import ReactModal from 'react-modal';
+import styles from './modal.module.scss';
+
+interface IModalProps {
+	isOpen: boolean;
+	elements: {
+		title: string;
+		body: string | React.ReactNode;
+		footer: React.ReactNode | null;
```
Comment (kyrylo-burmelov) [inline client/src/components/modals/modal/index.tsx:11]: you can add optional prop

```javascript
footer?: React.ReactNode;
```

File: client/src/components/modals/modal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+	return (
+		<ReactModal shouldFocusAfterRender={true} isOpen={isOpen} style={modalStyles}>
+			<div className={styles.modalContent}>
+				<div className={styles.header}>
+					<H3 className={styles.title}>{title}</H3>
+					<Icon icon="cross" className={styles.closeIcon} size={25} onClick={() => setIsOpen(false)} />
+				</div>
+				<div className={styles.body}>{body}</div>
+				{footer ? <div className={styles.footer}>{footer}</div> : null}
```
Comment (kyrylo-burmelov) [inline client/src/components/modals/modal/index.tsx:29]:
```
{footer && <div className={styles.footer}>{footer}</div>}
```

File: client/src/components/modals/modal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+		top: '50%',
+		left: '50%',
+		right: 'auto',
+		bottom: 'auto',
+		marginRight: '-50%',
+		transform: 'translate(-50%, -50%)',
+		padding: '0',
+		contentBoxSize: true,
+	},
+};
```
Comment (kyrylo-burmelov) [inline client/src/components/modals/modal/index.tsx:48]: move to config modal/config.ts

**Review by kyrylo-burmelov — CHANGES_REQUESTED** [review]
- You need to add import of modal component from 'components` folder. See examples from this folder
- We need to have a possibility to use other props from `react-modal` component. So, you need to add one one prop to `Modal` component. For example: `modalProps?: /*here TS interface from lib/*`.  And then spread it to modal component.
- We probably need a possibility to close the popup not only from modal component by clicking icon. Sometimes it's needed to close the  modal from the component from where modal was opened. Looks like it's not possible with your current implementation.
- Let's add possibility to open modal without close icon. When some action from modal is required for continue using application
- See comments below

**Review by kyrylo-burmelov — APPROVED** [review]
It would be better to add close icon for the header by default. And then add a possibility to hide it with additional props. But let's leave it like this for now

**Pull request merged**